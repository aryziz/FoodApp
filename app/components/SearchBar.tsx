import React from 'react';
import { View, TextInput, StyleSheet } from 'react-native';
import Feather from '@expo/vector-icons/Feather';

interface SearchBarProps {
    term: string;
    onTermChange: (newTerm: string) => void;
    onTermSubmit: () => void;
}

const SearchBar = (props: SearchBarProps) => {
    return (
        <View style={styles.backgroundStyle}>
            <Feather name="search" size={30} style={styles.iconStyle} testID="search-icon" />
            <TextInput
                autoCapitalize='none'
                autoCorrect={false}
                placeholder='Search'
                style={styles.inputStyle}
                value={props.term}
                onChangeText={props.onTermChange}
                onEndEditing={props.onTermSubmit}
                testID='searchInput'
            />
        </View>
    );
};

const styles = StyleSheet.create({
    backgroundStyle: {
        backgroundColor: '#F0EEEE',
        height: 50,
        borderRadius: 5,
        marginHorizontal: 15,
        flexDirection: 'row',
        marginTop: 10
    },
    inputStyle: {
        flex: 1,
        fontSize: 18
    },
    iconStyle: {
        fontSize: 35,
        alignSelf: 'center',
        marginHorizontal: 15
    }
});

export default SearchBar;