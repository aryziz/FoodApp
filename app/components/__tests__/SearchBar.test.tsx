import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import SearchBar from '../SearchBar';

jest.mock('expo-font');

describe('SearchBar', () => {
    it('renders correctly with placeholder text', async () => {
        const { getByPlaceholderText, getByTestId } = render(
            <SearchBar
                term=""
                onTermChange={() => {}}
                onTermSubmit={() => {}}
            />
        );

        await waitFor(() => {
            expect(getByPlaceholderText('Search')).toBeTruthy();
        });
        expect(getByTestId).toBeTruthy();
    });

    it('calls onTermChange when text is entered', async () => {
        const mockOnTermChange = jest.fn();
        const { getByPlaceholderText } = render(
            <SearchBar
                term=""
                onTermChange={mockOnTermChange}
                onTermSubmit={() => {}}
            />
        );

        const input = getByPlaceholderText('Search');
        fireEvent.changeText(input, 'Pizza');

        await waitFor(() => {
            expect(mockOnTermChange).toHaveBeenCalledWith('Pizza');
        });
    });

    it('should match JSON-yfied snapshot', () => {
        const tree = render(
            <SearchBar
                term=""
                onTermChange={() => {}}
                onTermSubmit={() => {}}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
