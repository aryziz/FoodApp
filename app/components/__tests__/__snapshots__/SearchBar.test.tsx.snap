// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match JSON-yfied snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#F0EEEE",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 50,
      "marginHorizontal": 15,
      "marginTop": 10,
    }
  }
>
  <Text />
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[Function]}
    onEndEditing={[Function]}
    placeholder="Search"
    style={
      {
        "flex": 1,
        "fontSize": 18,
      }
    }
    testID="searchInput"
    value=""
  />
</View>
`;
